File: AI.Core/OperatorsImplimentation/BasicOperators/Plus.cs
﻿using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    public class Plus : Operator
    {
        public Plus(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 2;
        }

        public override void Action()
        {
            var value = EnterContacts[0].Value + EnterContacts[1].Value;
            ExitContacts.SetValue(value);
        }
    }
}

File: NativeCore/OperatorsImplimentation/BasicOperators/Plus.cpp
#include "Plus.h"

using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Plus::Plus(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(2);
				}

				void Plus::Action()
				{
					auto value = EnterContacts[0]->Value + EnterContacts[1]->Value;
					ExitContacts->SetValue(value);
				}
			}
		}
	}
}
